lib/actions/write.js

```diff
@@ -1,104 +1,96 @@
-const axios = require('axios');
-const csv = require('csv');
-const _ = require('lodash');
-const { messages } = require('elasticio-node');
-const client = require('elasticio-rest-node')();
-const logger = require('@elastic.io/component-logger')();
+/* eslint-disable no-restricted-syntax,semi,comma-dangle,class-methods-use-this,
+no-param-reassign */
 
-const util = require('../util/util');
+const { AttachmentProcessor } = require('@elastic.io/component-commons-library')
+const { messages } = require('elasticio-node')
+const papa = require('papaparse')
 
-const REQUEST_TIMEOUT = process.env.REQUEST_TIMEOUT || 10000; // 10s
-const REQUEST_MAX_RETRY = process.env.REQUEST_MAX_RETRY || 7;
-const REQUEST_RETRY_DELAY = process.env.REQUEST_RETRY_DELAY || 7000; // 7s
-const REQUEST_MAX_CONTENT_LENGTH = process.env.REQUEST_MAX_CONTENT_LENGTH || 10485760; // 10MB
 const TIMEOUT_BETWEEN_EVENTS = process.env.TIMEOUT_BETWEEN_EVENTS || 10000; // 10s;
 
-let stringifier;
-let signedUrl;
-let timeout;
-let rowCount = 0;
-let ax;
-let putUrl;
+const rawData = []
+let timeout
 
-let readyFlag = false;
+async function proceedData(data, cfg) {
+  let csvString
+  const delimiter = cfg.separator ? cfg.separator : ','
 
-async function init(cfg) {
-  const delimiter = cfg.writer.separator || ',';
-  const header = cfg.includeHeaders !== 'No';
-  logger.trace('Using delimiter: \'%s\'', delimiter);
-  const options = {
-    header,
-    delimiter,
-  };
-
-  if (cfg.writer.columns) {
-    const columns = Object.keys(_.keyBy(cfg.writer.columns, 'property'));
-    options.columns = columns;
+  const unparseOptions = {
+    header: cfg.header,
+    delimiter
   }
 
-  stringifier = csv.stringify(options);
-  signedUrl = await client.resources.storage.createSignedUrl();
-  putUrl = signedUrl.put_url;
-  logger.trace('Uploading CSV file');
-  ax = axios.create();
-  util.addRetryCountInterceptorToAxios(ax);
-  readyFlag = true;
-}
-async function ProcessAction(msg, cfg) {
-  // eslint-disable-next-line consistent-this
-  const self = this;
-
-  while (!readyFlag) {
-    // eslint-disable-next-line no-loop-func,no-await-in-loop
-    await new Promise((resolve) => timeout(resolve, 100));
+  if (cfg.order) {
+    // create fields array from string
+    // eslint-disable-next-line
+    const fields = cfg.order.split(`${delimiter}`).map(x => { return x.trim() })
```

Comment:
There is a small bug here with regards to the having the delimter appear in a column name. However, I don't think we need to worry about fixing that now.

Reply:
We can force users to use something like double quotes for each header in this string or make using them as optional and try to find them by ourselves.

Reply:
I think the Papaparse library actually handles quote escaping quite nicely. I think that if you use `papaparse.parse()` on the `cfg.order` property, this problem should be solved.

Reply:
Yes, this solution works nice (also add trim to avoid unnecessary spaces)